test/Microsoft.TemplateEngine.Cli.UnitTests/TemplateResolutionTests/ListTemplateListResolverTests.cs

```diff
@@ -530,5 +530,216 @@ public void TestGetTemplateResolutionResult_NoMatch()
             Assert.False(matchResult.HasUnambiguousTemplateGroup);
             Assert.Equal(0, matchResult.UnambiguousTemplateGroup.Count);
         }
+
+
+
+        [Fact(DisplayName = nameof(TestGetTemplateResolutionResult_MatchByTags))]
+        public void TestGetTemplateResolutionResult_MatchByTags()
+        {
+            List<ITemplateInfo> templatesToSearch = new List<ITemplateInfo>();
+            templatesToSearch.Add(new TemplateInfo()
+            {
+                ShortName = "console",
+                Name = "Long name for Console App",
+                Identity = "Console.App.T1",
+                GroupIdentity = "Console.App.Test",
+                CacheParameters = new Dictionary<string, ICacheParameter>(),
+                Classifications = new List <string> {  "Common", "Test" },
+                Tags = new Dictionary<string, ICacheTag>(StringComparer.OrdinalIgnoreCase)
+                {
+                    { "language", ResolutionTestHelper.CreateTestCacheTag("L1") },
+                    { "type", ResolutionTestHelper.CreateTestCacheTag("project")}
+                },
+                BaselineInfo = new Dictionary<string, IBaselineInfo>()
+                {
+                    { "app", new BaselineInfo() },
+                    { "standard", new BaselineInfo() }
+                }
+            });
+
+
+            INewCommandInput userInputs = new MockNewCommandInput()
+            {
+                TemplateName = "Common",
+                IsListFlagSpecified = true,
+            };
+
+            ListOrHelpTemplateListResolutionResult matchResult = TemplateListResolver.GetTemplateResolutionResultForListOrHelp(templatesToSearch, new MockHostSpecificDataLoader(), userInputs, null);
+            Assert.True(matchResult.HasExactMatches);
+            Assert.Equal(1, matchResult.ExactMatchedTemplatesGrouped.Count);
+            Assert.Equal(1, matchResult.ExactMatchedTemplates.Count);
+            Assert.False(matchResult.HasLanguageMismatch);
+            Assert.False(matchResult.HasContextMismatch);
+            Assert.False(matchResult.HasBaselineMismatch);
+        }
+
+        [Fact(DisplayName = nameof(TestGetTemplateResolutionResult_MatchByTagsIgnoredOnNameMatch))]
+        public void TestGetTemplateResolutionResult_MatchByTagsIgnoredOnNameMatch()
+        {
+            List<ITemplateInfo> templatesToSearch = new List<ITemplateInfo>();
+            templatesToSearch.Add(new TemplateInfo()
+            {
+                ShortName = "console1",
+                Name = "Long name for Console App Test",
```

Comment:
Do we have tests for `HasClassificationMatchAndNameMismatch` method? I guess this test case would detect the error I mentioned above since:
- `Name` has a match with "Test"
- `ShortName` doesn't match with "Test"
- `HasClassificationMatchAndNameMismatch` returns a mismatch when it should have returned a match.

Reply:
This is exactly the test you've mentioned. 
`HasClassificationMatchAndNameMismatch ` is not even called in this case, as we have match by name in this case and condition `_coreMatchedTemplates.Where(t => t.IsMatch).Any()` is true in this case.

Reply:
Obviously, I don't fully understand this, but do we have a test that tests `HasClassificationMatchAndNameMismatch` method? If not, is it worthwhile to create one?

Reply:
I've added some tests for `HasClassificationMatchAndNameMismatch`, including the one for case above.  
I think they will be useful when implementing further changes to it, for example https://github.com/dotnet/templating/issues/2475.